src/loaders/lottie/tvgLottieBuilder.cpp

```diff
@@ -381,33 +381,46 @@ static void _repeat(LottieGroup* parent, Shape* path, RenderContext* ctx)
 }
 
 
-static void _appendRect(Shape* shape, float x, float y, float w, float h, float r, Matrix* transform, bool clockwise)
+static void _appendOffsetPath(Shape* shape, const PathCommand* cmds, uint32_t cmdsCnt, const Point* pts, uint32_t ptsCnt, const RenderOffsetPath* offsetPath, bool clockwise)
+{
+    Array<PathCommand> offsetCmds;
+    Array<Point> offsetPts;
+    _modifier(pts, ptsCnt, cmds, cmdsCnt, offsetCmds, offsetPts, offsetPath->offset * (clockwise ? 1.0f : -1.0f), offsetPath->join, offsetPath->miterLimit);
+
+    shape->appendPath(offsetCmds.data, offsetCmds.count, offsetPts.data, offsetPts.count);
+}
+
+
+static void _appendRect(Shape* shape, float x, float y, float w, float h, float r, Matrix* transform, const RenderOffsetPath* offsetPath, bool clockwise)
 {
     //sharp rect
     if (tvg::zero(r)) {
         PathCommand commands[] = {
             PathCommand::MoveTo, PathCommand::LineTo, PathCommand::LineTo,
-            PathCommand::LineTo, PathCommand::Close
+            PathCommand::LineTo, PathCommand::LineTo, PathCommand::Close
```

Comment:
Looks bad... Adding a drawing data for the unusual function seems a bad choice in general cases.

Reply:
on it

Reply:
[rectSamples.zip](https://github.com/user-attachments/files/16666744/rectSamples.zip)
![Aug-20-2024 00-56-41](https://github.com/user-attachments/assets/c84f5c8a-28d9-4e7d-b55b-8663c4b383ce)